extensions/multiply-divide/README.md

```diff
@@ -0,0 +1,62 @@
+# General design
+
+**Extension State: Under Development**  
+**Requires: Base, VWI, CP2.0**  
+**CPUID Bit: CP2.3**
+
+# Overview
+
+This extension adds multiplication, division, and remainder instructions. 
+
+# Added Instructions
+
+These instructions are in the expanded calculation opcode section of instructions.
+
+A `U` at the front of an instruction indicates that it treats its operands as unsigned.
+An `S` indicates that the operands are treated as signed.
+
+### Opcode table
+
+| `C CCCC CCCC` | NAME     | Operation                                  | Flags  | Comment |
+|---------------|----------|--------------------------------------------|--------|---------|
+| `0 0001 0000` | `UDIV`   | `A ← A / B`                                | None   | (1)     |
+| `0 0001 0001` | `SDIV`   | `A ← A / B`                                | None   | (1)     |
+| `0 0001 0010` | `UREM`   | `A ← A % B`                                | None   | (1) (2) |
+| `0 0001 0011` | `SREM`   | `A ← A % B`                                | None   | (1) (2) |
+| `0 0001 0100` | `UMUL`   | `A ← low(A × B)`                           | `CV`   | (3) (5) |
+| `0 0001 0101` | `SMUL`   | `A ← low(A × B)`                           | `CV`   | (3) (6) |                                          
+| `0 0001 0110` | `UHMUL`  | `A ← high(A × B)`                          | None   | (4)     |
+| `0 0001 0111` | `SHMUL`  | `A ← high(A × B)`                          | None   | (4)     |                                          
```

Comment:
```suggestion
| `0 0001 0000` | `UDIV`   | `A ← A / B`                                | `ZN`   | (1)     |
| `0 0001 0001` | `SDIV`   | `A ← A / B`                                | `ZN`   | (1)     |
| `0 0001 0010` | `UREM`   | `A ← A % B`                                | `ZN`   | (1) (2) |
| `0 0001 0011` | `SREM`   | `A ← A % B`                                | `ZN`   | (1) (2) |
| `0 0001 0100` | `UMUL`   | `A ← low(A × B)`                           | `CZN`   | (3) (5) |
| `0 0001 0101` | `SMUL`   | `A ← low(A × B)`                           | `CZN`   | (3) (6) |                                          
| `0 0001 0110` | `UHMUL`  | `A ← high(A × B)`                          | `ZN`   | (4)     |
| `0 0001 0111` | `SHMUL`  | `A ← high(A × B)`                          | `ZN`   | (4)     |                                          
```
The main justification for this is that it can simplify the ALU as mentioned above. There are a few reasons these might be useful anyway though.

- SREM setting `N` will allow for a more compact modulus instruction.
- Setting both `C` and `V` to the same value for UMUL and SMUL seems redundant, but I don't feel too strongly about it.

Reply:
(this change was discussed on Discord beforehand).